core/Network/TLS/Handshake/Client.hs

```diff
@@ -314,7 +311,7 @@ onServerHello ctx cparams sentExts (ServerHello rver serverRan serverSession cip
 
     -- intersect sent extensions in client and the received extensions from server.
     -- if server returns extensions that we didn't request, fail.
-    when (not $ null $ filter (not . flip elem sentExts . (\(ExtensionRaw i _) -> i)) exts) $
+    unless (null $ filter (not . flip elem sentExts . (\(ExtensionRaw i _) -> i)) exts) $
```

Comment:
Can't this be further simplified with `all`?

Reply:
Yes, we can. But I don't know whether or not @vincenthz likes it.

Reply:
I'm not a fan of `when not` -> `unless`, it's much less readable IMHO, but if you guys are happy with those changes, let's move forward. same goes for `all`. I either will adapt myself to the syntax or go mad :)

Reply:
Let's keep this as is.